### PR TITLE
feat(AP-5236): Introduce metrics utils app package

### PR DESCRIPTION
**Pull request by Drodevbar**
## Changes

Introduces the `metrics-utils` package to make working with Prom metrics easier and more standardized.

## Checklist

- [x] Apply one of following labels; `major`, `minor`, `patch` or `skip-release`
- [x] I've updated the documentation, or no changes were necessary
- [x] I've updated the tests, or no changes were necessary

### CODE REVIEW

File: packages/app/metrics-utils/package.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            "require": "./dist/index.cjs",
+            "import": "./dist/index.js"
+        }
+    },
+    "publishConfig": {
+        "access": "public"
+    },
+    "scripts": {
+        "build": "rimraf dist && vite build",
+        "dev": "vite watch",
```
Comment (kibertoad) [inline packages/app/metrics-utils/package.json:27]: this is not needed

File: packages/app/metrics-utils/package.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        "@lokalise/fastify-extras": "^24.3.1"
+    },
+    "devDependencies": {
+        "@biomejs/biome": "^1.9.2",
+        "@lokalise/biome-config": "^1.5.0",
+        "@lokalise/package-vite-config": "3.1.0",
+        "@vitest/coverage-v8": "^2.1.1",
+        "rimraf": "^6.0.1",
+        "typescript": "5.6.2",
+        "vitest": "^2.0.3"
```
Comment (kibertoad) [inline packages/app/metrics-utils/package.json:48]: let's use latest vitest

File: packages/app/metrics-utils/src/AbstractCounterMetric.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  helpDescription: string
+  label: TMetricLabel
+  measurementKeys: TMetricMeasurementKeys
+}
+
+export abstract class AbstractCounterMetric<
+  TMetricLabel extends string,
+  TMetricMeasurementKeys extends string[],
+> {
+  private readonly counterMetric: Counter<TMetricLabel> | null
```
Comment (kibertoad) [inline packages/app/metrics-utils/src/AbstractCounterMetric.ts:19]: can you add comment explaining it's null when metrics are disabled on app level?

File: packages/app/metrics-utils/package.json [hunk 4]
```diff
@@ -0,0 +1,50 @@
+{
+    "name": "@lokalise/metrics-utils",
```
Comment (kibertoad) [inline packages/app/metrics-utils/package.json:2]: can you also add readme, explaining how to use and what it does?

File: packages/app/metrics-utils/package.json (hunk 1 above)
Comment (Drodevbar) [reply packages/app/metrics-utils/package.json]: Adjusted to the format of other packages across this repo ✅

File: packages/app/metrics-utils/package.json (hunk 2 above)
Comment (Drodevbar) [reply packages/app/metrics-utils/package.json]: Updated `vitest` and `vite` to latest stable versions ✅

File: packages/app/metrics-utils/src/AbstractCounterMetric.ts (hunk 3 above)
Comment (Drodevbar) [reply packages/app/metrics-utils/src/AbstractCounterMetric.ts:20]: Comment added ✅

File: packages/app/metrics-utils/package.json (hunk 4 above)
Comment (Drodevbar) [reply packages/app/metrics-utils/package.json:2]: I enhanced the readme with usage instructions. Could you please see if it's clearer now from your point of view?

File: packages/app/metrics-utils/README.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
+# Metrics Utils
+
+This package contains set of utilities to work with Prometheus metrics.
+
+Table of contents:
+1. [AbstractCounterMetric](#AbstractCounterMetric)
+
+### AbstractCounterMetric
+
+A base class for single counter-like metrics.
```
Comment (kibertoad) [inline packages/app/metrics-utils/README.md:10]: can you add short explanation why you may prefer this over raw Prometheus metrics? e. g. safe multiple inits

Comment (Drodevbar) [reply packages/app/metrics-utils/README.md:10]: Sure, added an extra explanation there

**Review by kibertoad — APPROVED** [review]
Thank you, this is awesome!

**Comment by Drodevbar** [comment]
@kibertoad It looks like the addition of a new package somehow conflicts with deps of another one (`context-fastify-plugins`) - https://github.com/lokalise/shared-ts-libs/actions/runs/11382381650/job/31665741626 I'll investigate. Sharing with you now in case you have some bright ideas how to solve it.

**Comment by kibertoad** [comment]
@Drodevbar probably not strictly enough pinned deps are messing things up. I can check in an hour or so

**Comment by Drodevbar** [comment]
@kibertoad I see it's fastify-metrics which requires fastify bo be >= 5 (as part of `metrics-utils`), while `context-fastify-plugins` only works with fastify v4. Do you think we should update fastify-plugins to work with v5 of fastify or use package aliases, workspaces, etc.?

**Pull request merged**